addons/pupswap/pupswap.lua

```diff
@@ -0,0 +1,125 @@
+local settings = require('settings')
+local automaton = require('automaton')
+
+local command = require('core.command')
+
+local defaults = {
+    equip_delay = 0.5,
+    sets = {}
+}
+local options = settings.load(defaults)
+
+command.arg.register('set_name', '<set_name:string>')
+
+-- Addon command Handlers
+local ps = command.new('ps')
+
+local equip_set = function(set_name)
+    local set = options.sets[set_name]
+    if not automaton.activated then
+        coroutine.schedule(function()
+            automaton:remove_all()
+
+            --TODO: inform user start of set equipping
+            coroutine.sleep(options.equip_delay)
+            if automaton.validate_head_name(set.head) then
```

Comment:
I would remove all validation from the equip function. I think it makes more sense to validate it on option parsing only. That way you can tell people immediately that they have invalid data and they won't find out mid-battle when they try to equip a set. Additionally it makes the execution code faster and cleaner.

There's an event you can subscribe to called `settings.settings_change`, it triggers any time settings are reloaded. That would be a good thing to register and use for set validation.

Reply:
I like that a lot, should I error if I find an invalid set? these feels like a right time to error, since it would happen on load.